index.js

```diff
@@ -63,13 +62,18 @@ const handleArguments = (file, args, options = {}) => {
 };
 
 const handleOutput = (options, value, error) => {
-	if (typeof value !== 'string' && !Buffer.isBuffer(value)) {
+	if (typeof value !== 'string' && !ArrayBuffer.isView(value)) {
 		// When `execaSync()` errors, we normalize it to '' to mimic `execa()`
 		return error === undefined ? undefined : '';
 	}
 
 	if (options.stripFinalNewline) {
-		return stripFinalNewline(value);
+		if (typeof value === 'string') {
+			return stripFinalNewline(value);
+		}
+
+		const rawStripedString = stripFinalNewline(new TextDecoder().decode(value));
+		return new TextEncoder().encode(rawStripedString);
```

Comment:
Decoding and re-encoding the `Uint8Array` to/from a string might be rather slow on big outputs.

Since `\n` and `\r` are always single UTF-8 bytes, we should be able to strip them by directly manipulating the `Uint8Array`.

Actually, it seems like [the code from `stripFinalNewline()`](https://github.com/sindresorhus/strip-final-newline/blob/main/index.js) should work with `Uint8Array`, not only `Buffer` (although I am not sure this is intentional).

However, that code does call `.slice()` once (twice on Windows), which requires copying the whole underlying `ArrayBuffer`, which can be slow too. 

A more efficient approach that does not involve any copying could be:

```js
const stripTypedArrayNewline = (value) => value[value.length - 1] === LF
  ? value.subarray(0, value[value.length - 2] === CR ? -2 : -1)
  : value

const LF = '\n'.charCodeAt();
const CR = '\r'.charCodeAt();
```

What do you think?

Reply:
Quick benchmarks running 1e6 iterations, comparing:
  1. Using `.subarray()`
  2. Using `.slice()` with `strip-final-newline`
  3. Using `TextEncoder`/`TextDecoder` + `strip-final-newline`

For:
  - Unix newlines on a small input: 32ms (1), 40ms (2), 452ms (3)
  - Windows newlines on a small input: 31ms (1), 77ms (2), 477ms (3)
  - Unix newlines on a big input: 32ms (1), 1120ms (2), 2233ms (3)
  - Windows newlines on a big input: 31ms (1), 2378ms (2), 2705ms (3)

Small input is `hello\n` or `hello\n\r`, running 1e6 iterations.
Big input is the same but repeated 1e6 times, with 1e3 iterations.

Reply:
Update: I have added a PR to `strip-final-newline` directly: https://github.com/sindresorhus/strip-final-newline/pull/5